File: front-end/src/application/usecase/operation/operation-get-use-case.ts
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Operation } from '../../../domain/value-object/operation';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { OperationName } from '@/domain/value-object/operation-name';

export interface OperationGetUseCase {
    /**
     * Get an operation by name
     * @param operationName
     * @returns operation selected
     */
    getOperation(operationName: OperationName): Promise<Either<DataError, Operation>>;

    /**
     * Get all operations
     * @returns all operations
     */
    getAllOperations(): Promise<Either<DataError, Operation[]>>;
}

File: back-end/src/main/java/es/ull/project/application/usecase/operation/OperationGetUseCase.java
package es.ull.project.application.usecase.operation;

import java.util.List;
import java.util.Optional;

import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;

public interface OperationGetUseCase {
    /**
     * Get an operation by Name
     * 
     * @param operation
     * @return operation selected
     */
    Optional<Operation> getOperation(OperationName operationName);

    /**
     * Get all operations
     * 
     * @return all operations
     */
    List<Operation> getAllOperations();
}
